openml-lightgbm/lightgbm-provider/src/main/java/com/feedzai/openml/provider/lightgbm/LightGBMUtils.java

```diff
@@ -91,13 +118,30 @@ static CpuArchitecture getCpuArchitecture(final String cpuArchName) {
      */
     static private void loadSharedLibraryFromJar(
             final String sharedLibResourceName,
-            final CpuArchitecture cpuArchitecture
+            final CpuArchitecture cpuArchitecture,
+            final boolean isAlpine
     ) throws IOException {
 
-        logger.debug("Loading LightGBM shared lib: {} for {}.", sharedLibResourceName, cpuArchitecture);
+        if (isAlpine && cpuArchitecture.getLgbmNativeLibsFolder().equals("arm64")) {
+            throw new IOException("Trying to use LightGBM on an unsupported architecture arm64 on Alpine.");
+        }
+
+        final String libraryPath;
+        if (cpuArchitecture.getLgbmNativeLibsFolder().equals("amd64") && isAlpine) {
```

Comment:
You can compare enums using cpuArchitecture == CpuArchitecture.AARCH64...

Reply:
Thanks! I changed it in the latest commit.